Foundation/NSPathUtilities.swift

```diff
@@ -550,13 +567,61 @@ extension NSString {
     }
     
     public func getFileSystemRepresentation(_ cname: UnsafeMutablePointer<Int8>, maxLength max: Int) -> Bool {
+#if os(Windows)
+        let fsr = UnsafeMutablePointer<WCHAR>.allocate(capacity: max)
+        defer { fsr.deallocate() }
+        guard _getFileSystemRepresentation(fsr, maxLength: max) else { return false }
+        let str = String(decodingCString: fsr, as: UTF16.self)
+        return str.withCString() {
+            let chars = strnlen_s($0, max)
+            guard chars < max else {
+                return false
+            }
+            cname.assign(from: $0, count: chars + 1)
+            return true
+        }
+#else
         guard self.length > 0 else {
             return false
         }
-        
         return CFStringGetFileSystemRepresentation(self._cfObject, cname, max)
```

Comment:
Would it not be better to just fix `CFStringGetFileSystemRepresentation` to work for Windows as well, then it should work for other callers inside `CoreFoundation`?

Reply:
Do we think it's appropriate to have the Windows implementation do path normalization? The POSIX implementation doesn't, it converts to a cstring in the appropriate system encoding (UTF8/UTF16), but doesn't do any path normalization itself.

In this case, by path normalization, I mean dropping leading and trailing slashes when appropriate (as below), then optionally passing the path to GetFullPathNameW to fully normalize it.

cc @millenomi